draft-ietf-httpbis-messaging-latest.xml

```diff
@@ -1903,6 +1904,12 @@ https://www.example.org
    It is unknown whether the reset problem is exclusive to TCP or might also
    be found in other transport connection protocols.
 </t>
+<t>
+   Note that some servers will interpret the client closing its side of a
+   connection as a loss of interest in the response, and therefore close
+   their side without sending any outstanding portions of the response,
+   in order to save resources.
+</t>
```

Comment:
I don't think we should add this just because one implementation fails to correctly interpret the protocol. I don't care if it is in Go, they can and should fix the bug.

Reply:
Wait, that's a bit confusing anyway. 

Servers do interpret a full close (client closing read side) as a connection termination, which is fine. What they don't do, normally, is mark the whole connection closed if the client sends FIN (half-close) because that doesn't mean anything other than no more requests to read.

Reply:
Reading the issue again I think this is probably the best we can do. I rearranged some; PTAL.